src/server/bucket_guard.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2024, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+#include "util/fibers/synchronization.h"
+
+namespace dfly {
+
+struct CondVarWithBoolean {
```

Comment:
This looks very generic. 
we can call it sturct ConditionFlag with bool flag and cond_var members. 
note sturct members should not end with _

The Guard name will be ConditionGuard